File: tdd_demo/src/main/resources/static/fizzbuzz.js
// fizzbuzz.js
(function(exports) {
  "use strict";

  function FizzBuzz(number) {
    this.number = number || 0;
  }
  exports.FizzBuzz = FizzBuzz;

  FizzBuzz.prototype = {
    toString: function() {
      if (this.number % 3 == 0) {
    	  return "fizz";
      } else {
      	return "" + this.number;
      }
    }
  };

})(this);

File: tdd_demo/src/main/java/com/connect_group/tdd_demo/examples/fizzbuzz/FizzBuzz.java
package com.connect_group.tdd_demo.examples.fizzbuzz;

public class FizzBuzz {

	public String fizzBuzz(int number) {
		if (number % 15 == 0) {
			return "fizz buzz";
		} else if (number % 3 == 0) {
			return "fizz";
		} else if (number % 5 == 0) {
			return "buzz";
		} else {
			return new Integer(number).toString();
		}
	}
}
